xds/credentials/credentials.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package credentials supports registering and retreiving credentials for xDS
+// communications.
+//
+// Notice: This package is EXPERIMENTAL and may be changed or removed in a later
+// release.
+package credentials
```

Comment:
How about this?

```go
package xds/bootstrap

type Credentials interface {
	Build(json.RawMessage) (credentials.Bundle, error)
	Name() string
}

func RegisterCredentials(Credentials)
func GetCredentials(string) Credentials
```

Reply:
Makes sense, moved the registry inside the bootstrap package.

Reply:
You moved to `xds/internal/xdsclient/bootstrap`.  What I was thinking was a new package in `xds/bootstrap`, so it could be used by others.  Is this what you were thinking @menghanl?

Reply:
Sorry for the confusion, I misunderstood the suggestion. Created a new bootstrap package under xds. Let me know if this looks better.